File: model_1/icpc-apac-2024-tasks/banhbo_1_solution.cpp
#include <iostream>
using namespace std;

const long long MOD = 998244353;

long long mod_exp(long long base, long long exp, long long mod) {
    long long result = 1;
    base %= mod;
    while (exp > 0) {
        if (exp & 1) {
            result = (result * base) % mod;
        }
        base = (base * base) % mod;
        exp >>= 1;
    }
    return result;
}

int main() {
    long long r, c;
    cin >> r >> c;
    long long exponent = (long long)r * c;
    cout << mod_exp(2, exponent, MOD) << endl;
    return 0;
}

File: model_1/icpc-apac-2024-tasks/banhbo_1_solution.py
MOD = 998244353

def main():
    import sys
    r, c = map(int, sys.stdin.readline().split())
    a = r // 6
    b = c // 7
    exponent = 42 * a * b
    result = pow(2, exponent, MOD)
    print(result)

if __name__ == '__main__':
    main()